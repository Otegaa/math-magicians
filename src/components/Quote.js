import React, { useState, useEffect } from 'react';

const Quote = () => {
  const [quote, setQuote] = useState([]);
  const [isLoading, setIsLoading] = useState(false);
  const [isError, setIsError] = useState(false);

  useEffect(() => {
    const getQuotes = async () => {
      setIsLoading(true);
      setIsError(false);
      try {
        const res = await fetch(
          'https://api.api-ninjas.com/v1/quotes?category=learning',
          {
            method: 'GET',
            headers: {
              'X-Api-Key': 'RHDLyf4DRjsSHUBxCTge4Q==rSin3VvTVUJbdujG',
            },
            contentType: 'application/json',
          },
        );
        const quote = await res.json();
        setQuote(quote);
      } catch (error) {
        setIsError(true);
      }
      setIsLoading(false);
    };
    getQuotes();
  }, []);

  return (
    <>
      {isError && <h1>Something went wrong! Try again</h1>}
      {isLoading ? (
        <h3 className="quote">Loading...</h3>
      ) : (
        <ul className="quote">
          {quote.map((q) => {
            const { quote } = q;
            return (
              <li key={quote} className="quote-list">
                {quote}
              </li>
            );
          })}
        </ul>
      )}
    </>
  );
};

export default Quote;
