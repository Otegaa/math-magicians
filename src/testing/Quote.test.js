import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import Quote from '../pages/Quote';

describe('Quotes', () => {
  it('Should render correctly', () => {
    const { getByText } = render(<Quote />);
    expect(getByText('Loading...')).toBeInTheDocument();
  });
  it('renders correctly', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
