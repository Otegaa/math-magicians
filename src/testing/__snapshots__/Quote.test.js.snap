// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes renders correctly 1`] = `
<ul
  className="quote"
/>
`;
